Add ColorPicker and Handle components

adamxberlin opened the pull request:
Partial implementation of #2

Commit:
Update and clean up package.json

Commit:
Add info and clean up template files

Commit:
Rename files and remove unused code

Commit:
Add ColorPicker and Handle components

sunyatasattva commented on src/components/handle.tsx:
```diff
@@ -0,0 +1,63 @@
+
+    const boundingRect = this.container.current.getBoundingClientRect();
+
+    const delta = {
```
Yes! Love the way you cleaned that up 👍

adamxberlin replied:
Thx 🙏😄

sunyatasattva commented on src/components/handle.tsx:
```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+const handleRotation = Symbol();
+const toggleMouseDown = Symbol();
```
Why would you use `Symbol`s here? It really seems like a completely forced, overly-clever solution, to the wrong problem. Unless I'm missing something, there is no reason to use those over normal method names.

If you wanted privacy, note that `Symbol`s are not private. If you wanted to denote privacy, Typescript allows you to have `private` methods that make that explicit. Or, if you wanted, you could use the official `#method()` experimental proposal through Babel. But I would just stay with TS.

adamxberlin replied:
https://medium.com/@martin_hotell/10-typescript-pro-tips-patterns-with-or-without-react-5799488d6680 ⬅️Bullet point 3

Just trying out new patterns 😬 But I agree, it feels a bit convoluted... We can change it back to private to be more explicit if you want. Come to think of it, I think I'd prefer to actually.

I mean, the private accessor is just to help us avoid mistakes while coding in this case . Everything on runtime happens client-side, so we don't really care if the end user messes around with the code.

adamxberlin replied:
Fixed and pushed it to the PR branch 👍

sunyatasattva reviewed (commented):
Everything besides my one comment about `Symbol` looks great! However, this does not technically close #2 as you have written in your comment, because #2 is the whole Color Picker component, which we don't yet have.

adamxberlin commented:
> Everything besides my one comment about `Symbol` looks great! However, this does not technically close #2 as you have written in your comment, because #2 is the whole Color Picker component, which we don't yet have.

You're right! Changed the comment to "Partial implementation of #2" so it's more correct 👍

Commit:
Change private method syntax

sunyatasattva committed:
Connect handle rotation to color

sunyatasattva committed:
Refactor Color Picker in several ways. Closes #2

1. Color Picker doesn't use a gradient anymore, instead switched approach to `canvas` for
more accurate rendition of color wheel hues.
2. Allows for multiple colors and handles to be selected
3. Allows to specify an inner and outer radius
4. Emits events with the currently selected colors
